### DISCUSSION TITLE
Pulling Xray scan status from Jfrog Artifactory

### DISCUSSION THREAD

**Discussion by giet12aei033** [community/community#134917] [Programming Help]
### Select Topic Area

Question

### Body

Hi All,

I am having workflow where i am building docker image and pushing it to jfrog artifactory. Now, i want to pull the image from jfrog artifactory once xray scan is completed and push to ecr repo. could someone please help me out?

>       - name: Login to Artifactory
>         uses: docker/login-action@v3
>         with:
>           registry: ${{ env.ARTIFACTORY_REPO }}.${{ env.ARTIFACTORY_NAME}}
>           username: xxxxxxxxx
>           password: ${{ env.JF_DOCKER_ACCESS_TOKEN }} 
>           
>       - name: Push Docker Image to Artifactory
>         working-directory: ${{ inputs.BUILD_LOCATION }}
>         run: |
>           docker tag ${{ env.ARTIFACTORY_REPO }}/${{inputs.ECR_NAME }}:${{ github.run_number }} ${{ env.ARTIFACTORY_REPO }}.${{ env.ARTIFACTORY_NAME}}/${{inputs.ECR_NAME }}:${{ github.run_number }}
>           docker push ${{ env.ARTIFACTORY_REPO }}.${{ env.ARTIFACTORY_NAME}}/${{inputs.ECR_NAME }}:${{ github.run_number }}

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @giet12aei033 !

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the [GitHub product Discussions](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/about-discussions). This question should be in the Programming Help category. I've gone ahead and moved it for you. Good luck!

**Comment by geniuszly** [comment]
To accomplish the workflow where you pull a Docker image from JFrog Artifactory after an Xray scan is completed and then push it to an Amazon ECR repository, you can try the following:
Use JFrog's REST API to query the status of the Xray scan,for example:
```bash
curl -u <user>:<apikey> -X GET "https://<artifactory_url>/xray/api/v1/summary/artifact/<artifact_path>"
```
Once the scan status is confirmed as successful (usually with a `PASSED` status), proceed to pull the image from Artifactory to pull the image:
```bash
docker pull ${{ env.ARTIFACTORY_REPO }}.${{ env.ARTIFACTORY_NAME}}/${{inputs.ECR_NAME }}:${{ github.run_number }}
```
Then push it to the ECR repository:
```bash
aws ecr get-login-password --region <region> | docker login --username AWS --password-stdin <aws_account_id>.dkr.ecr.<region>.amazonaws.com
docker tag ${{ env.ARTIFACTORY_REPO }}.${{ env.ARTIFACTORY_NAME}}/${{inputs.ECR_NAME }}:${{ github.run_number }} <aws_account_id>.dkr.ecr.<region>.amazonaws.com/${{inputs.ECR_NAME }}:${{ github.run_number }}
docker push <aws_account_id>.dkr.ecr.<region>.amazonaws.com/${{inputs.ECR_NAME }}:${{ github.run_number }}
```
You can integrate API calls and conditional steps into your GitHub Actions workflow to automate this process.
